### ISSUE TITLE
Incorrect possible dereference of a null reference after assertion method

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#32335]
**Version Used**: 

Microsoft Visual Studio Enterprise 2019 Preview
Version 16.0.0 Preview 1.1
VisualStudio.16.Preview/16.0.0-pre.1.1+28408.50
Microsoft .NET Framework
Version 4.7.03056

**Steps to Reproduce**:

Compile the following code

```cs
#nullable enable

using System.Runtime.CompilerServices;

public class C
{
  public extern void M([EnsuresNotNull] C? c1);

  public void Test1(C? c1, C? c2, C? c3)
  {
    M(c1?.Method());

    c1.ToString(); // warning
  }
    
  public extern C? Method();
}

namespace System.Runtime.CompilerServices
{
  class EnsuresNotNullAttribute : System.Attribute { }
}
```

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BiAdgVwBtCJhC4ACOXU8gWAChGABABguYEYA6AJX1wwAlmDjcAwgHswAByHkEAZUQA3IQGM4UANyMWAZg4AmCuMYBvRhQ6G4AD3gJcHACwUAsgAoA2gFFcUPgIWgBykjAhRIQAuqYA/BTqnACUukwM1syGzG4AKlownJ7iCUlI8YlG5SWJ+slWFJYZ1h6eSXHc7nAwABaSKJ7JqXrN1kncuZKKMAhCuADmg9oUAPQrFADuEE5z8w0Avg3WDVlUDojONV29/UuMh+k0olAyEJocPPyCImJSsvKIZQINSaKAWE4mfyBYJQMIRKIAQRgMyEwHw8AoIA+3CRKLRGPMFAeDyAA==

**Expected Behavior**:

No warnings. `c1?.Method()` is known to be not null and consequently `c1` is known to be not null.

**Actual Behavior**:
Incorrect warning
`warning CS8602: Possible dereference of a null reference.`

**Comment by jcouv** [comment]
Verified this was fixed in 16.1p1 (or earlier).
